x/crosschain/keeper/keeper_cross_chain_tx_vote_outbound_tx.go

```diff
@@ -105,32 +103,20 @@ func (k msgServer) VoteOnObservedOutboundTx(goCtx context.Context, msg *types.Ms
 	// FinalizeOutbound sets final status for a successful vote
 	// FinalizeOutbound updates CCTX Prices and Nonce for a revert
 
-	err = func() error { //err = FinalizeOutbound(k, ctx, &cctx, msg, ballot.BallotStatus) // remove this line
+	tmpCtx, commit := ctx.CacheContext()
+	err = func() error { //err = FinalizeOutbound(k, ctx, &cctx, msg, ballot.BallotStatus)
 		cctx.GetCurrentOutTxParam().OutboundTxObservedExternalHeight = msg.ObservedOutTxBlockHeight
-		zetaBurnt := cctx.InboundTxParams.Amount
-		zetaMinted := cctx.GetCurrentOutTxParam().Amount
 		oldStatus := cctx.CctxStatus.Status
 		switch ballot.BallotStatus {
 		case observerTypes.BallotStatus_BallotFinalized_SuccessObservation:
 			switch oldStatus {
 			case types.CctxStatus_PendingRevert:
-				cctx.CctxStatus.ChangeStatus(&ctx, types.CctxStatus_Reverted, "", cctx.LogIdentifierForCCTX())
+				cctx.CctxStatus.ChangeStatus(types.CctxStatus_Reverted, "")
 			case types.CctxStatus_PendingOutbound:
-				cctx.CctxStatus.ChangeStatus(&ctx, types.CctxStatus_OutboundMined, "", cctx.LogIdentifierForCCTX())
+				cctx.CctxStatus.ChangeStatus(types.CctxStatus_OutboundMined, "")
 			}
-
 			newStatus := cctx.CctxStatus.Status.String()
-			if zetaBurnt.LT(zetaMinted) {
-				// TODO :Handle Error ?
-			}
-			balanceAmount := zetaBurnt.Sub(zetaMinted)
-			if cctx.GetCurrentOutTxParam().CoinType == common.CoinType_Zeta { // TODO : Handle Fee for other coins
-				err := HandleFeeBalances(k, ctx, balanceAmount)
```

Comment:
This should be removed. The movement of fees is already handled in inbound tx observation.